File: java/src/main/java/mx/iteso/test/MilkIceCreamStoreTest.java
package mx.iteso.test;


import mx.iteso.Cone;
import mx.iteso.Stores.MilkIceCreamStore;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Class for water ice cream tests.
 */
public class MilkIceCreamStoreTest {
    /**
     * testing Guava ice  Cream.
     */
    @Test
    public void testGuavaMilkIceCream() {
        Cone cone = new MilkIceCreamStore().prepareCone("milk", "Guava");
        assertEquals(cone.getIceCream().getFlavor(), "Guava milk ice cream");
    }

    /**
     * testing Mango ice  Cream.
     */
    @Test
    public void testMangoMilkIceCream() {
        Cone cone = new MilkIceCreamStore().prepareCone("milk", "Mango");
        assertEquals(cone.getIceCream().getFlavor(), "Mango milk ice cream");
    }

    /**
     * testing Strawberry ice  Cream.
     */
    @Test
    public void testStrawberryMilkIceCream() {
        Cone cone = new MilkIceCreamStore().prepareCone("milk", "Strawberry");
    assertEquals(cone.getIceCream().getFlavor(), "Strawberry milk ice cream");
    }

    /**
     * testing Pineapple ice  Cream.
     */
    @Test
    public void testPineappleMilkIceCream() {
        Cone cone = new MilkIceCreamStore().prepareCone("milk", "Pineapple");
    assertEquals(cone.getIceCream().getFlavor(), "Pineapple milk ice cream");
    }
}

File: csharp/factorypattern/test/stores/MilkIceCreamStoreTest.cs
using factorypattern.main;
using factorypattern.main.stores;
using NUnit.Framework;

namespace factorypattern.test.stores
{
    internal class MilkIceCreamStoreTest
    {
        private MilkIceCreamStore _milkIceCreamStore;
        private Cone _cono;

        [SetUp]
        public void setUp()
        {
            _milkIceCreamStore = new MilkIceCreamStore();
            _cono = new Cone();
        }

        [TestCase]
        public void yogurtStoreTest()
        {
            _cono = _milkIceCreamStore.prepareCone("Mango");
            Assert.AreEqual("Mango Milk Ice Cream", _cono.getIceCream().getName());
            _cono = _milkIceCreamStore.prepareCone("Pineapple");
            Assert.AreEqual("Pineapple Milk Ice Cream", _cono.getIceCream().getName());
            _cono = _milkIceCreamStore.prepareCone("Strawberry");
            Assert.AreEqual("Strawberry Milk Ice Cream", _cono.getIceCream().getName());
            _cono = _milkIceCreamStore.prepareCone("Guava");
            Assert.AreEqual("Guava Milk Ice Cream", _cono.getIceCream().getName());

        }
    }
}

